clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/caching/view/model/KubernetesV2SecurityGroup.java

```diff
@@ -0,0 +1,160 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.clouddriver.kubernetes.v2.caching.view.model;
+
+import com.netflix.spinnaker.cats.cache.CacheData;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.caching.Keys;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.caching.agent.KubernetesCacheDataConverter;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesKind;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesManifest;
+import com.netflix.spinnaker.clouddriver.model.SecurityGroup;
+import com.netflix.spinnaker.clouddriver.model.SecurityGroupSummary;
+import com.netflix.spinnaker.clouddriver.model.securitygroups.Rule;
+import io.kubernetes.client.models.V1NetworkPolicy;
+import io.kubernetes.client.models.V1NetworkPolicyPort;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+@EqualsAndHashCode(callSuper = true)
+@Data
+@Slf4j
+public class KubernetesV2SecurityGroup extends ManifestBasedModel implements SecurityGroup {
+  KubernetesManifest manifest;
+  Keys.InfrastructureCacheKey key;
+  String id;
+
+  @Override
+  public String getApplication() {
+    return getMoniker().getApp();
+  }
+
+  Set<Rule> inboundRules;
+  Set<Rule> outboundRules;
+
+  @Override
+  public SecurityGroupSummary getSummary() {
+    return KubernetesV2SecurityGroupSummary.builder()
+        .id(id)
+        .name(id)
+        .build();
+  }
+
+  KubernetesV2SecurityGroup(KubernetesManifest manifest, String key, Set<Rule> inboundRules, Set<Rule> outboundRules) {
+    this.manifest = manifest;
+    this.id = manifest.getName();
+    this.key = (Keys.InfrastructureCacheKey) Keys.parseKey(key).get();
+    this.inboundRules = inboundRules;
+    this.outboundRules = outboundRules;
+  }
+
+  public static KubernetesV2SecurityGroup fromCacheData(CacheData cd) {
+    if (cd == null) {
+      return null;
+    }
+
+    KubernetesManifest manifest = KubernetesCacheDataConverter.getManifest(cd);
+
+    if (manifest == null) {
+      log.warn("Cache data {} inserted without a manifest", cd.getId());
+      return null;
+    }
+
+    Set<Rule> inboundRules = new HashSet<>();
+    Set<Rule> outboundRules = new HashSet<>();
+
+    if (manifest.getKind() != KubernetesKind.NETWORK_POLICY) {
+      log.warn("Unknown security group kind " + manifest.getKind());
```

Comment:
If this `warn` is hit, this function still returns a security group with no inbound or outbound rules. Is that OK or should these be more than warnings?

Reply:
Yup, this warning is a signal that the logic needs to be implemented. Returning `null` would be surprising since at least an object with a `manifest` is expected to be returned.